File: frontend/src/types/supplier.ts
/** 供应商状态 */
export type SupplierStatus = 'draft' | 'pending' | 'active' | 'rejected' | 'disabled' | 'cancelled'
export type SupplierQualificationExpiryStatus = 'valid' | 'near_expire' | 'expired'

/** 供应商资质文件 */
export interface SupplierQualificationFile {
  id: number
  name: string
  size: string
  url?: string | null
}

/** 供应商信息 */
export interface Supplier {
  id: number
  supplierCode: string
  supplierName: string
  contactName: string
  contactPhone: string
  contactEmail: string
  address: string
  /** 供应商类型（单选） */
  supplierType: string
  /** 社会信用代码 */
  unifiedCreditCode: string
  /** 银行账号 */
  bankAccount: string
  /** 开户行 */
  bankName: string
  licenseNo: string
  licenseExpiresAt: string | null
  licenseExpiryStatus?: SupplierQualificationExpiryStatus | null
  licenseRemainingDays?: number | null
  foodLicenseNo: string
  foodLicenseExpiresAt: string | null
  foodLicenseExpiryStatus?: SupplierQualificationExpiryStatus | null
  foodLicenseRemainingDays?: number | null
  creditScore: number
  scoreQualification: number | null
  scoreQuality: number | null
  scorePrice: number | null
  scoreDelivery: number | null
  scoreUpdatedAt?: string | null
  scoreStatisticsPeriod?: string | null
  aiLevel?: string | null
  recommendPriority?: string | null
  riskWarningLevel?: 'low' | 'medium' | 'high' | null
  scoreQualitySampleInsufficient?: boolean
  scorePriceSampleInsufficient?: boolean
  scoreDeliverySampleInsufficient?: boolean
  optimizationSuggestions?: string[]
  status: SupplierStatus
  disableReason: string | null
  cancelReason: string | null
  qualificationFiles: SupplierQualificationFile[]
  auditAt: string | null
  auditRemark: string | null
  createdByName?: string | null
  updatedByName?: string | null
  auditByName?: string | null
  tenantId: number
  createdAt: string
  updatedAt: string
}

/** 供应商表单 */
export interface SupplierForm {
  supplierCode: string
  supplierName: string
  contactName: string
  contactPhone: string
  contactEmail: string
  address: string
  /** 供应商类型（单选） */
  supplierType: string
  /** 社会信用代码 */
  unifiedCreditCode: string
  /** 银行账号 */
  bankAccount: string
  /** 开户行 */
  bankName: string
  licenseNo: string
  licenseExpiresAt: string | null
  foodLicenseNo: string
  foodLicenseExpiresAt: string | null
  status: SupplierStatus
  disableReason?: string | null
  qualificationFiles: SupplierQualificationFile[]
}

export interface SupplierDisablePayload extends SupplierForm {
  reason: string
}

export interface SupplierCancelPayload {
  reason: string
}

/** 供应商查询参数 */
export interface SupplierQuery {
  keyword?: string
  status?: SupplierStatus | ''
}

/** 供应商统计 */
export interface SupplierStatistics {
  total: number
  activeCount: number
  pendingCount: number
  nearExpireCount: number
}

/** 供应商导入失败明细 */
export interface SupplierImportFailure {
  rowNum: number
  supplierCode: string
  supplierName: string
  supplierTypeRawValue?: string | null
  unifiedCreditCode: string
  documentType?: string | null
  documentNo?: string | null
  failedField: string
  errorMessage: string
}

/** 供应商导入结果 */
export interface SupplierImportResult {
  total: number
  successCount: number
  failCount: number
  hasErrors: boolean
  errorFileUrl: string | null
  failures: SupplierImportFailure[]
}

/** 供应商唯一性校验结果 */
export interface SupplierDuplicateCheckResult {
  supplierCodeDuplicate: boolean
  supplierCodeMessage: string | null
  supplierNameDuplicate: boolean
  supplierNameMessage: string | null
  licenseNoDuplicate: boolean
  licenseNoMessage: string | null
  foodLicenseNoDuplicate: boolean
  foodLicenseNoMessage: string | null
}

/** 供应商导入预校验冲突明细 */
export interface SupplierImportValidationConflict {
  rowNum: number
  field: string
  conflictValue: string | null
  message: string
}

/** 供应商导入预校验结果 */
export interface SupplierImportValidationResult {
  effectiveRowCount: number
  rowLimitExceeded: boolean
  duplicateConflicts: SupplierImportValidationConflict[]
}

File: backend/scm-service/src/main/java/com/xykj/scm/entity/Supplier.java
package com.xykj.scm.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 供应商实体
 * 对应数据库表: scm_supplier
 */
@Data
@TableName("scm_supplier")
public class Supplier implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private String supplierCode;
    private String supplierName;
    private String contactName;
    private String contactPhone;
    private String contactEmail;
    private String address;

    private String supplierType;
    /**
     * 社会信用代码
     */
    private String unifiedCreditCode;
    private String bankAccount;
    private String bankName;

    private String licenseNo;
    private LocalDateTime licenseExpiresAt;
    private String foodLicenseNo;
    private LocalDateTime foodLicenseExpiresAt;

    /**
     * JSON 字符串（数组）
     */
    private String qualificationFiles;

    private BigDecimal creditScore;
    private BigDecimal scoreQualification;
    private BigDecimal scoreQuality;
    private BigDecimal scorePrice;
    private BigDecimal scoreDelivery;

    private String status;
    private String disableReason;
    private String cancelReason;
    private Long auditBy;
    private LocalDateTime auditAt;
    private String auditRemark;

    private Long orgId;
    private Long tenantId;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}
